src/browser/ui/ReactMount.js

```diff
@@ -370,8 +370,21 @@ var ReactMount = {
     var reactRootElement = getReactRootElementInContainer(container);
     var containerHasReactMarkup =
       reactRootElement && ReactMount.isRenderedByReact(reactRootElement);
+    var containerHasNonRootReactChild =
+      ReactMount.hasNonRootReactChild(container);
 
-    var shouldReuseMarkup = containerHasReactMarkup && !prevComponent;
+    if (__DEV__) {
+      warning(
+        !containerHasNonRootReactChild,
+        'renderComponent(...): Replacing React-rendered children with a new ' +
+        'root component.'
```

Comment:
Can we add a "If you intended to update the children of this node, you should instead have the component that rendered the existing children update its state and render the new components instead of calling the top-level React.render to update a subtree."

Also, s/renderComponent/render/. :)